### PR TITLE
Deep queries on metachain

### PR DESCRIPTION
**Pull request by andreibancioiu**
- Handle creation of `chain handler` for SQ Query Service (metachain vs. shard).
 - Remove `OneTestNetwork` - replace it with `MiniNetwork`.
 - Added / adjusted some tests.

### CODE REVIEW

File: factory/api/apiResolverFactory.go [hunk 1]
```diff
@@ -356,9 +358,18 @@ func createScQueryElement(
 		return nil, errDecode
 	}
 
-	apiBlockchain, err := blockchain.NewBlockChain(disabled.NewAppStatusHandler())
-	if err != nil {
-		return nil, err
+	var apiBlockchain data.ChainHandler
+
+	if isMetachain {
```
Comment (iulianpascalau) [inline factory/api/apiResolverFactory.go:363]: I would extract this in a small function so we can drop the else branch :)

File: integrationTests/testNetwork.go [hunk 2]
```diff
@@ -34,7 +34,7 @@ type GasScheduleMap = map[string]map[string]uint64
 // TestNetwork wraps a set of TestProcessorNodes along with a set of test
 // Wallets, instantiates them, controls them and provides operations with them;
 // designed to be used in integration tests.
-// TODO combine TestNetwork with the preexisting TestContext and OneNodeNetwork
+// TODO combine TestNetwork with the preexisting TestContext and MiniNetwork
```
Comment (andreibancioiu) [inline integrationTests/testNetwork.go:37]: Only in the end I've seen that we have a `TestNetwork` struct. Can we postpone the migration to `TestNetwork` for another PR, or should we do it now?

File: integrationTests/vm/wasm/queries/queries_test.go [hunk 3]
```diff
@@ -100,79 +98,64 @@ func TestQueries(t *testing.T) {
...
+	historyOfGetState[4] = getState(t, network.ShardNode, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
+	historyOfGetNow[4] = queryHistoryGetNow(t, network.ShardNode, scAddress, core.OptionalUint64{HasValue: true, Value: 4})
 
 	require.Equal(t, snapshotsOfGetState[1], historyOfGetState[1])
 	require.Equal(t, snapshotsOfGetNow[1].blockNonce, historyOfGetNow[1].blockNonce)
-	// This does not seem right!
-	require.Equal(t, block4.GetRootHash(), historyOfGetNow[1].stateRootHash)
 
 	require.Equal(t, snapshotsOfGetState[2], historyOfGetState[2])
 	require.Equal(t, snapshotsOfGetNow[2].blockNonce, historyOfGetNow[2].blockNonce)
```
Comment (andreibancioiu) [inline integrationTests/vm/wasm/queries/queries_test.go:117]: Removed some assertions (see delta).

File: factory/api/apiResolverFactory_test.go [hunk 4]
```diff
@@ -448,5 +449,24 @@ func TestCreateApiResolver_createScQueryElement(t *testing.T) {
 		require.True(t, strings.Contains(strings.ToLower(err.Error()), "hasher"))
 		require.Nil(t, scQueryService)
 	})
+}
+
+func TestCreateApiResolver_createBlockchainForScQuery(t *testing.T) {
```
Comment (andreibancioiu) [inline factory/api/apiResolverFactory_test.go:454]: Not very proud of this unit test.

Question for review: keep it or remove it?

File: integrationTests/testNetwork.go (hunk 2 above)
Comment (iulianpascalau) [reply integrationTests/testNetwork.go:37]: we can postpone

File: factory/api/apiResolverFactory_test.go [hunk 5]
```diff
@@ -448,5 +449,24 @@ func TestCreateApiResolver_createScQueryElement(t *testing.T) {
...
+	t.Parallel()
+
+	t.Run("for metachain", func(t *testing.T) {
+		t.Parallel()
 
+		apiBlockchain, err := api.CreateBlockchainForScQuery(core.MetachainShardId)
+		require.NoError(t, err)
+		require.Equal(t, "*blockchain.metaChain", fmt.Sprintf("%T", apiBlockchain))
```
Comment (iulianpascalau) [inline factory/api/apiResolverFactory_test.go:462]: :+1: 
This actually tests the factory characteristic of the function. It serves the purpose well.

File: factory/api/apiResolverFactory_test.go (hunk 4 above)
Comment (iulianpascalau) [reply factory/api/apiResolverFactory_test.go:454]: It is ok. I would keep the tests :D

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**